### DISCUSSION TITLE
Use with-editor for magit-run and friends

### DISCUSSION THREAD

**Discussion by david-greene-cb** [magit/magit#5107] [Ideas]
It would be nice to use with-editor to executed the commands prefixed by `!`.  This way any scripts that might invoke something interactive like a rebase would work naturally.  Alternatively, new suffix commands could provide the same functionality as existing suffix commands (e.g. `!!`) but run them using with-editor.

**Comment by tarsius** [comment]
> It would be nice to use with-editor to executed the commands prefixed by `!`.

I don't understand where you want to do this. In a terminal maybe? Maybe you are looking for `with-editor-async-shell-command` and similar commands? Please provide an example (not of a command that you might want to invoke, but of how you want to invoke it).

**Reply by david-greene-cb** [reply]
`!! git <my git plugin>`

Where `<my git plugin>` does a rebase.

Yes, `with-editor-async-shell-command` will do what I want, without the benefit of switching to the project root first.

**Reply by tarsius** [reply]
The part that I did not understand is that "the commands *prefixed* by `!`" means "the commands available from `magit-run` (bound to `!`)". (Though admittedly when you mentioned "suffix commands (e.g. `!!`)", I should have made the connection. (I don't really us `magit-run`, so `!` and `!!` didn't trigger any memory.)

**Comment by tarsius** [comment]
Done in 28bcd29db547ab73002fb81b05579e4a2e90f048.
